src/eat.c

```diff
@@ -58,7 +58,8 @@ static int tin_ok(struct obj *);
    they can be sacrificed regardless of age which implies that they never
    become rotten */
 #define nonrotting_corpse(mnum) \
-    ((mnum) == PM_LIZARD || (mnum) == PM_LICHEN || is_rider(&mons[mnum]) \
+    ((mnum) == PM_LIZARD || (mnum) == PM_LICHEN || \
+     (mnum >= LOW_PM && is_rider(&mons[mnum])) \
```

Comment:
Oh, also, mnum should be in parentheses in case the macro is called on some more complicated thing that could theoretically complicate the order of operations I guess.